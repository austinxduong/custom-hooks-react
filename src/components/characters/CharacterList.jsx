import React, { useState } from 'react';
import { Link } from 'react-router-dom';
import Avatar from './Character';

import { useCharacters } from '../../state/character';

const CharacterList = () => {
  const [page, setPage] = useState(1);
  const { characters, loading } = useCharacters(page);
  if(loading) return <h1>en route...</h1>;

  const characterElements = characters.map((character) => (
    <li key={character._id}>
      <Link to={`/${character._id}`}>
        <Avatar {...character} />
      </Link>
    </li>
  ));

  return (
    <>
      <button
        disabled ={page <= 1}
        onClick={() => setPage((prevPage) => prevPage - 1)}>
                    &lt;
      </button>
      {page}
      <button onClick={() => setPage((prevPage) => prevPage + 1)}>&gt;</button>
      <ul>{characterElements}</ul>
    </>
  );
};

export default CharacterList;



