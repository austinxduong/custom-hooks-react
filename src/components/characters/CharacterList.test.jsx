import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';


import CharacterList from './CharacterList';

describe.only('Avatar component aka Character.jsx', () => {
  it('renders individual character from Avatar component', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <CharacterList
       
        />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
