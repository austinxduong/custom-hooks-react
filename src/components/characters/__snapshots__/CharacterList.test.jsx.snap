// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar component aka Character.jsx renders individual character from Avatar component 1`] = `
<DocumentFragment>
  <h1>
    en route...
  </h1>
</DocumentFragment>
`;
